src/main/java/slimeknights/tconstruct/tools/modifiers/ability/interaction/BlockTransformModifier.java

```diff
@@ -121,6 +124,8 @@ public InteractionResult afterBlockUse(IToolStackView tool, ModifierEntry modifi
             totalTransformed++;
             didTransform = true;
 
+            BlockTransformModifierHook.afterTransformBlock(tool, context, newTarget, newPos, action);
```

Comment:
This placement is inconsistent. The above call to the hook runs serverside only, but this call potentially could run on the client if the targeted block did not transform. Could lead to really weird bugs with the code consuming this hook that are really hard to reproduce.

Just move the call to after the client side hook check. I am undecided on if you should damage the tool before or after this hook runs, check what I did for other hooks like harvest and make it consistent.